renku_notebooks/api/notebooks.py

```diff
@@ -495,9 +501,32 @@ def _on_error(error_msg):
             )
         except requests.exceptions.ConnectionError as exc:
             _on_error(f"User secrets storage service could not be contacted {exc}")
+        else:
+            if response.status_code != 201:
+                _on_error(f"User secret could not be created {response.json()}")
+
+    # NOTE: Create a secret for each storage that has saved secrets
+    for cloud_storage in storages:
+        if cloud_storage.secrets and cloud_storage.base_name:
+            request_data = {
+                "name": f"{cloud_storage.base_name}-secrets",
+                "namespace": server.k8s_client.preferred_namespace,
+                "secret_ids": list(cloud_storage.secrets),
+                "owner_references": [owner_reference],
+                "key_mapping": cloud_storage.secrets,
+            }
 
-        if response.status_code != 201:
-            _on_error(f"User secret could not be created {response.json()}")
+            try:
+                response = requests.post(
+                    config.user_secrets.secrets_storage_service_url + "/api/secrets/kubernetes",
```

Comment:
suggestion: As this is used in 2 places now, we should make the path some constant defined maybe at the top of the file?

Reply:
I've refactored the duplicate code into a function.